test/helpers/functional/fluentd.go

```diff
@@ -0,0 +1,113 @@
+package functional
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/openshift/cluster-logging-operator/internal/pkg/generator/forwarder"
+	"github.com/openshift/cluster-logging-operator/pkg/certificates"
+	"github.com/openshift/cluster-logging-operator/pkg/constants"
+	"github.com/openshift/cluster-logging-operator/pkg/logger"
+	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	"github.com/openshift/cluster-logging-operator/test/builder"
+	"github.com/openshift/cluster-logging-operator/test/helpers/oc"
+)
+
+var (
+	maxDuration time.Duration
+)
+
+type FluentdFunctionalFramework struct {
+	Name      string
+	Namespace string
+	Conf      string
+	Error     error
+	image     string
+	labels    map[string]string
+	clfYaml   string
+}
+
+func init() {
+	maxDuration, _ = time.ParseDuration("2m")
+}
+
+func NewFluentdFunctionalFramework() *FluentdFunctionalFramework {
+	testName := fmt.Sprintf("test-fluent-%d", rand.Intn(1000))
+	framework := &FluentdFunctionalFramework{
+		Name:      testName,
+		Namespace: testName,
+		image:     utils.GetComponentImage(constants.FluentdName),
+		labels: map[string]string{
+			"testtype": "functional",
+			"testname": testName,
+		},
+	}
+	return framework
+}
+
+//WithLogForwarder supplies ClusterLogForwarder for generating a config in the form of YAML
+func (f *FluentdFunctionalFramework) WithLogForwarder(yaml string) *FluentdFunctionalFramework {
+	f.clfYaml = yaml
+	return f
+}
+func (f *FluentdFunctionalFramework) RunCommand(cmdString string) (string, error) {
+	cmd := strings.Split(cmdString, " ")
+	logger.Infof("running command: %s", cmd)
+	return oc.Exec().Pod(f.Name).WithNamespace(f.Name).WithCmd(cmd[0], cmd[1:]...).RunFor(maxDuration)
+}
+
+//Deploy the objects needed to functional test
+func (f *FluentdFunctionalFramework) Deploy() error {
+	if f.Error = oc.Literal().From(fmt.Sprintf("oc create ns %s", f.Namespace)).Output(); f.Error != nil {
+		return f.Error
+	}
+	//generate empty config
+	if f.Conf, f.Error = forwarder.Generate(f.clfYaml, false); f.Error != nil {
+		return f.Error
+	}
+	if f.Error = certificates.GenerateCertificates(f.Namespace,
+		utils.GetScriptsDir(), "elasticsearch",
+		utils.DefaultWorkingDir); f.Error != nil {
+		return f.Error
+	}
+	if _, f.Error = builder.NewConfigMapBuilder(f.Namespace, f.Name).
+		Add("fluent.conf", f.Conf).
+		Add("run.sh", string(utils.GetFileContents(utils.GetShareDir()+"/fluentd/run.sh"))).
+		Create(); f.Error != nil {
+		return f.Error
+	}
+	certsName := "certs-" + f.Name
+	if _, f.Error = builder.NewConfigMapBuilder(f.Namespace, certsName).
+		Add("tls.key", string(utils.GetWorkingDirFileContents("system.logging.fluentd.key"))).
+		Add("tls.crt", string(utils.GetWorkingDirFileContents("system.logging.fluentd.crt"))).
+		Create(); f.Error != nil {
+		return f.Error
+	}
+
+	podBuilder := builder.NewPodBuilder(f.Namespace, f.Name).
+		WithLabels(f.labels).
+		AddConfigMapVolume("config", f.Name).
```

Comment:
Add Volume 
```
- hostPath:
    path: /var/log
    type: ""
  name: varlog
```
and volumeMount
```
volumeMounts:
- mountPath: /var/log
  name: varlog
```

Reply:
The point is we shouldnt need to mount any node level directories for the test.  This would allow us to directly inject messages into the container and remove the setup without having artifacts on the host

Reply:
How do we plan to inject logs into fluentd's pipeline? i.e. what will be the `<source>` ?

Reply:
we can still define a source to the same path but jump onto the container via exec and echo something in the required format or something like that , maybe a shared init container

Reply:
Just FYI: The image has fluent-cat so you could use that to inject messages directly into a `forward` source. You could also echo/cat stuff into a file that fluent is tailing as you point out. Check test/helpers/cmd package has a Reader type that can read from pod files, you could create equivalent Writer types to write to pod files or via fluent-cat direct to fluentd.